src/librustc_borrowck/diagnostics.rs

```diff
@@ -314,6 +314,76 @@ let c = &i; // still ok!
 ```
 "##,
 
+E0501: r##"
+This error indicates that a mutable variable which has been captured in a
+closure is being used while it is still captured by the closure. Because the
+closure has borrowed the variable, the variable is not available for use until
+the closure goes out of scope.
+
+Example of erroneous code:
+```compile_fail
+struct Sample;
+fn inside_closure(x: &mut Sample) {
+    // Actions which require unique access
+}
+fn outside_closure(x: &mut Sample) {
+    // Actions which require unique access
+}
+fn foo(a: &mut Sample){
+    let mut bar = || {
+        inside_closure(a)
+    };
+    outside_closure(a); // error: cannot borrow `*a` as mutable because previous
+                        //        closure requires unique access.
+}
+```
+
+To fix this error, you can place the closure in it's own environment:
```

Comment:
"its" vs "it's" is even covered on the first page you linked. Trust me here.

On Sat, Apr 30, 2016 at 2:34 PM, Timothy McRoy <notifications@github.com>
wrote:

> In src/librustc_borrowck/diagnostics.rs
> <https://github.com/rust-lang/rust/pull/33294#discussion_r61673172>:
>
> > +fn inside_closure(x: &mut Sample) {
> > +    // Actions which require unique access
> > +}
> > +fn outside_closure(x: &mut Sample) {
> > +    // Actions which require unique access
> > +}
> > +fn foo(a: &mut Sample){
> > +    let mut bar = || {
> > +        inside_closure(a)
> > +    };
> > +    outside_closure(a); // error: cannot borrow `*a` as mutable because previous
> > +                        //        closure requires unique access.
> > +}
> > +```
> > +
> > +To fix this error, you can place the closure in it's own environment:
>
> No apostrophe indicates plurality, I believe.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/rust/pull/33294/files/c177b1490d5538b515e49b1fd2d8529e9887e2a8#r61673172>
>